### PR TITLE
Design Spec for LR Routing Queue model

### PR DESCRIPTION
**Pull request by hisundar**
## Overview

Description:

Details design based on discussions and can be useful when looking back to see why certain decisions were made.

Why should this be merged: 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [ ] There are no TODOs left in the code
- [ ] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [ ] Change is covered by automated tests
- [ ] Public API has Javadoc

### CODE REVIEW

File: runtime/proto/queue.proto [hunk 1]
```diff
@@ -14,4 +14,10 @@ message CorfuGuidMsg {
 
 message CorfuQueueMetadataMsg {
     int64 txSequence = 1;
+}
+
+// This message forms the value part of the payload used for LogReplication over Routing Queues
+message RoutingTableEntryMsg {
+    string destinations = 1;
```
Comment (souravgarg) [inline runtime/proto/queue.proto:21]: destinations field marked as repeated above?

Comment (pankti-m) [reply runtime/proto/queue.proto:21]: +1

Comment (hisundar) [reply runtime/proto/queue.proto]: ack - addressed

**Review by pankti-m — APPROVED** [review]

File: docs/log-replicator/routing_queues.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+LR Server on source and sink will have one session per remote destination.
+Data from the clients will be written into the following well known Routing Queues
+1. `LRQ_Send_LogEntries`: This is the one shared queue on the sender where delta updates are placed transactionally.
+2. `LRQ_Send_SnapSync`: This is the one shared queue on the sender where full snapshot sync updates are placed.
+3. `LRQ_Recv_<remote_id>`: This is the queue where the LR on Sink places the updates arriving from the remote source.
+
+Each update made carries destinations and is also tagged with a destination specific stream tag with the following convention:
+1. `lrq_logentry_<remote_id>`: Destination specific stream tag applied when entries are placed into `LRQ_Send_LogEntries`
+2. `lrq_snapsync_<remote_id>`: Destination specific stream tag applied when entries are placed into `LRQ_Send_SnapSync`
+3. `lrq_recv`: This is the tag applied on the sink by LR server to all updates be it full sync or snapshot sync.
```
Comment (shreayp) [inline docs/log-replicator/routing_queues.md:39]: "full sync or snapshot sync" -> "log entry sync or snapshot sync"?

Comment (hisundar) [reply docs/log-replicator/routing_queues.md:39]: thank you - yes i meant log entry sync the second time

**Review by shreayp — APPROVED** [review]

**Pull request merged**